src/account/types/mod.rs

```diff
@@ -228,3 +230,58 @@ impl TryFrom<u32> for CoinType {
         Ok(coin_type)
     }
 }
+
+// Custom serialization to return the coin type number
+impl Serialize for CoinType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            CoinType::IOTA => serializer.serialize_u32(self.clone() as u32),
+            CoinType::Shimmer => serializer.serialize_u32(self.clone() as u32),
+        }
```

Comment:
Why use a match statement if the actions undertaken are the same for both options?

Reply:
Good point, you're right shouldn't be a match there
Initially I provided the matching coin type number hardcoded, so there it wasn't the same, but didn't remove the match after changing it, will do now